Source/ACE.Server/WorldObjects/Player_Spells.cs

```diff
@@ -71,7 +71,7 @@ public void HandleActionMagicRemoveSpellId(uint spellId)
                 return;
             }
 
-            ChangesDetected = true;
+            if (!DoNotSave) { ChangesDetected = true; }
```

Comment:
ChangesDetected should still be set, and shouldn't be conditional on DoNotSave

Reply:
I forgot to remove this - it was added in some of my early testing.